tests/reverse/test_deleting_related.py

```diff
@@ -0,0 +1,176 @@
+import io
+import PIL
+
+from decimal import Decimal as D
+
+from django.core.files import File
+from django.test import TestCase
+
+from oscar.core.loading import get_model
+
+from oscar_odin.mappings.catalogue import products_to_db
+from oscar_odin.resources.catalogue import (
+    Product as ProductResource,
+    Image as ImageResource,
+    ProductClass as ProductClassResource,
+    Category as CategoryResource,
+)
+from oscar_odin.mappings.constants import (
+    ALL_CATEGORY_FIELDS,
+    ALL_PRODUCTIMAGE_FIELDS,
+    ALL_STOCKRECORD_FIELDS,
+)
+
+Product = get_model("catalogue", "Product")
+ProductClass = get_model("catalogue", "ProductClass")
+ProductAttribute = get_model("catalogue", "ProductAttribute")
+Category = get_model("catalogue", "Category")
+Partner = get_model("partner", "Partner")
+Stockrecord = get_model("partner", "Stockrecord")
+
+
+class DeleteRelatedModelReverseTest(TestCase):
```

Comment:
Are these all scenario's you can think of for tests? I think we need some more tests to be honest.